packages/components/tests/integration/components/hds/form/text-input/base-test.js

```diff
@@ -0,0 +1,90 @@
+
+module('Integration | Component | hds/form/text-input/base', function (hooks) {
+  setupRenderingTest(hooks);
+
+  hooks.afterEach(() => {
+    resetOnerror();
+  });
+
+  test('it renders the component', async function (assert) {
+    await render(hbs`<Hds::Form::TextInput::Base id="test-form-text-input" />`);
+    assert.dom('#test-form-text-input').exists();
+  });
+  test('it should render with a CSS class that matches the component name', async function (assert) {
+    await render(hbs`<Hds::Form::TextInput::Base id="test-form-text-input" />`);
+    assert.dom('#test-form-text-input').hasClass('hds-form-text-input');
+  });
+
+  // TYPE
+
+  test('it should render the "text" type if no type is declared', async function (assert) {
+    await render(hbs`<Hds::Form::TextInput::Base id="test-form-text-input" />`);
+    assert.dom('#test-form-text-input').hasAttribute('type', 'text');
+  });
+  test('it should render the correct type depending on the @type prop', async function (assert) {
+    await render(
+      hbs`<Hds::Form::TextInput::Base @type="email" id="test-form-text-input" />`
+    );
+    assert.dom('#test-form-text-input').hasAttribute('type', 'email');
+  });
+
+  // VALUE
+
+  test('it should render the input with the value provided via @value argument', async function (assert) {
+    await render(
+      hbs`<Hds::Form::TextInput::Base @value="abc123" id="test-form-text-input" />`
+    );
+    assert.dom('#test-form-text-input').hasValue('abc123');
+  });
+
+  // INVALID
+
+  test('it should render the correct CSS class if the @isInvalid prop is declared', async function (assert) {
+    await render(
+      hbs`<Hds::Form::TextInput::Base id="test-form-text-input" @isInvalid={{true}} />`
+    );
+    assert
+      .dom('#test-form-text-input')
+      .hasClass('hds-form-text-input--is-invalid');
+  });
+
+  // WIDTH
+
+  test('it should render the input with a fixed width if a @width value is passed', async function (assert) {
+    await render(
+      hbs`<Hds::Form::TextInput::Base @width="248px" id="test-form-text-input" />`
+    );
+    assert.dom('#test-form-text-input').hasStyle({ width: '248px' });
+  });
+
+  // ATTRIBUTES
+
+  test('it should spread all the attributes passed to the component', async function (assert) {
+    assert.expect(3);
+    await render(
+      hbs`<Hds::Form::TextInput::Base id="test-form-text-input" class="my-class" data-test1 data-test2="test" />`
+    );
+    assert.dom('#test-form-text-input').hasClass('my-class');
+    assert.dom('#test-form-text-input').hasAttribute('data-test1');
+    assert.dom('#test-form-text-input').hasAttribute('data-test2', 'test');
+  });
+
+  // ASSERTIONS
+
+  test('it should throw an assertion if an incorrect value for @type is provided', async function (assert) {
+    const errorMessage =
+      '@type for "Hds::Form::TextInput" must be one of the following: text, email, password, url, search, date, time, received: foo';
```

Comment:
[suggestion] in the future if we were to change the list of input types we'll have to update this error message which feels a little too tightly coupled. for the purpose of this test we care that an error is captured, not necessarily what the exact message is. what do you think?

Reply:
Yes, but that's on purpose. I could import (maybe?) the list from the component JS, but let's imagine I make an error there (eg. deleting one type), then the test would automatically auto-fix itself, and the error go unnoticed. Makes sense my thinking?

Reply:
I think my suggestion is more that asserting that _any_ error is thrown is sufficient and what the actual contents of the error message are is not particularly high value

Reply:
Ahh. I don't think there was a different way to test assertion (there is almost zero literature on this). I am not sure how to test assertions with just the "throw" part. Do you have ideas/suggestions?

Reply:
I'd need to do some digging. I think it's safe to table my suggestion for now and revisit if/when this is a problem.